src/GravityForms/GravityForms.php

```diff
@@ -30,41 +30,99 @@ public function GFFormSubmission(array $entry, array $form)
     {
         $this->setSupplier($form);
 
-        // if there is no supplier set just return the form.
         if (empty($this->supplier)) {
             return $form;
         }
 
         try {
-            $result = $this->createZaak($entry, $form);
+            $zaak = $this->createZaak($entry, $form);
+
+            if (! $zaak instanceof Zaak) { // Fallback.
+                throw new Exception('Het verwachte resultaat na het aanmaken van een Zaak voldoet niet.');
```

Comment:
Cryptische foutmelding wel 😄

Reply:
Klopt, dit is het bericht dat de burger ziet. Suggesties zijn welkom ;)

Reply:
In het template die ingeladen wordt staat dit:

```
            <div class="content">
                <div id="readspeaker">
                    <p class="mb-1">
                        Het aanmaken van uw zaak is mislukt.
                    </p>
                    <?php if (! empty($vars['error'])) : ?>
                        <p>Gebruik het volgende bericht in uw correspondentie met ons: "<?= $vars['error']; ?>"</p>
                    <?php endif; ?>
                </div>
            </div>
```
In deze context vind ik de huidige melding meer zinvol. Hoewel 'Probeer het opnieuw of neem contact op.' ook goed gebruikt kan worden om de plek in de code te vinden waar we moeten zijn. Wellicht in dit geval beter om specifieker te zijn?

Reply:
Aaah dat had ik niet gezien. Dan ben ik het met je eens dat de huidige melding meer zinvol is. 

Ik zit alleen te kijken wanneer de `$zaak` variabele geen `Zaak` is. De [`addZaak()`](https://github.com/OpenWebconcept/plugin-owc-gravityforms-zaaksysteem/blob/c17b0dba6e2fdc3601ef0979851adfcacb4d11d1/src/Clients/DecosJoin/Actions/CreateZaakAction.php#L21) method heeft een nullable `Zaak` returntype, maar als ik de method bekijk is er niets wat `null` teruggeeft. De [`create()`](https://github.com/OpenWebconcept/plugin-owc-gravityforms-zaaksysteem/blob/c17b0dba6e2fdc3601ef0979851adfcacb4d11d1/src/Endpoints/ZakenEndpoint.php#L107) method op de `ZakenEndpoint` geeft altijd een `Zaak` terug. Als ik het nu zo zie zou die check er uit kunnen en kan de returntype in van `addZaak()` zonder nullable modifier

Reply:
Goed punt, volgens mij is dit nog iets ouds waarbij er wel eens een NULL door ons zelf werd gegooid. Ik heb hem aangepast.